Source/cg_PresetsManager.cpp

```diff
@@ -92,97 +92,94 @@ bool PresetsManager::forceLoad(int const presetNumber)
+                auto const inversedNormalizedElevation{ static_cast<float>(presetData.getDoubleAttribute(elevation)) };
+                snapshot.z = MAX_ELEVATION * (1.0f - inversedNormalizedElevation);
+            }
+        }
+        if (source.isPrimarySource()) {
+            snapshots.primary = snapshot;
         } else {
-            terminalPosition = snapshots.primary.position;
+            snapshots.secondaries.add(snapshot);
         }
-        mSources.getPrimarySource().setPosition(terminalPosition, Source::OriginOfChange::presetRecall);
-
-        if (mSources.getPrimarySource().getSpatMode() == SpatMode::cube) {
-            Radians elevation;
-            if (presetData->hasAttribute(zTerminalPositionId)) {
-                auto const inversedNormalizedTerminalElevation{ static_cast<float>(
-                    presetData->getDoubleAttribute(zTerminalPositionId)) };
-                elevation = MAX_ELEVATION * (1.0f - inversedNormalizedTerminalElevation);
-            } else {
-                elevation = snapshots.primary.z;
-            };
+    }
 
-            mSources.getPrimarySource().setElevation(elevation, Source::OriginOfChange::presetRecall);
-        }
+    // load the snapshots into the enforcers, which are references to the ControlGrisAudioProcessor members
+    mPositionLinkEnforcer.loadSnapshots(snapshots);
+    if (mSources.getPrimarySource().getSpatMode() == SpatMode::cube) {
+        mElevationLinkEnforcer.loadSnapshots(snapshots);
     }
-    mLastLoadedPreset = presetNumber;
 
-    // send a change message, which will end up calling SourceLinkEnforcer::enforceSourceLink()
-    // to position the secondary sources
-    sendChangeMessage();
+    auto const xTPositionId{ getFixedPosSourceName(FixedPositionType::terminal, SourceIndex{ 0 }, 0) };
+    auto const yTPositionId{ getFixedPosSourceName(FixedPositionType::terminal, SourceIndex{ 0 }, 1) };
+    auto const zTPositionId{ getFixedPosSourceName(FixedPositionType::terminal, SourceIndex{ 0 }, 2) };
+
+    // position the first source
+    juce::Point<float> terminalPosition;
+    if (presetData.hasAttribute(xTPositionId) && presetData.hasAttribute(yTPositionId)) {
+        juce::Point<float> const inversedNormalizedTerminalPosition{
+            static_cast<float>(presetData.getDoubleAttribute(xTPositionId)),
+            static_cast<float>(presetData.getDoubleAttribute(yTPositionId))
+        };
+        auto const inversedTPosition{ inversedNormalizedTerminalPosition * 2.0f - juce::Point<float>{ 1.0f, 1.0f } };
+        terminalPosition = juce::Point<float>{ inversedTPosition.getX(), inversedTPosition.getY() * -1.0f };
+    } else {
+        terminalPosition = snapshots.primary.position;
+    }
+    mSources.getPrimarySource().setPosition(terminalPosition, Source::OriginOfChange::presetRecall);
+
+    if (mSources.getPrimarySource().getSpatMode() == SpatMode::cube) {
```

Comment:
I wonder if these `if` shouldn't become `switch` so that we get "unused enumeration value" warnings if one day there is another mode being added that requires some special handling there